Tribler/Main/metadata-injector.py

```diff
@@ -81,7 +82,13 @@ def initialize(self):
 
         self._logger.info(u"Starting session...")
         self.session = Session(sscfg)
-        self.session.prestart()
+        self.session.initialize_database()
+
+        failed, has_to_upgrade = self.session.has_to_upgrade_database()
+        if has_to_upgrade:
+            self.session.upgrade_database()
+        elif failed:
+            self.session.stash_database()
```

Comment:
Hmmm this piece of code is duplicate a few times. Maybe add create a function in session for it?